eclipse-theia/theia
# Add enable state of agent to preferences

#### What it does

The selected model was already persisted in the preferences,
but the enablement state was not.
Here the settings were extended to include the enablement state.
The new setting is consumed and updated from the AgentService.

#### How to test



Enable/Disable an agent and reload the page and check the agent has the same state of enablement as before the reload.

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
